When the first line in the server's response is an empty line, lftp treats it
as if it were the first header and expects more, when it should probably just
fall into its HTTP 0.9 code path.

--- lftp/src/Http.cc	2003-12-12 17:43:32.000000000 -0500
+++ lftp/src/Http.cc	2003-12-12 17:43:27.000000000 -0500
@@ -1067,7 +1067,7 @@
 	 const char *eol=find_eol(buf,len,&eol_size);
 	 if(eol)
 	 {
-	    if(eol==buf)
+	    if((eol==buf) && status)
 	    {
 	       DebugPrint("<--- ","",4);
 	       recv_buf->Skip(eol_size);
